src/Database/Relations/BelongsToMany.php

```diff
@@ -1,10 +1,147 @@
-<?php namespace Winter\Storm\Database\Relations;
+<?php
 
+namespace Winter\Storm\Database\Relations;
+
+use Illuminate\Database\Eloquent\Builder;
 use Illuminate\Database\Eloquent\Relations\BelongsToMany as BelongsToManyBase;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Database\Eloquent\Model;
+use Winter\Storm\Database\Pivot;
 
-class BelongsToMany extends BelongsToManyBase
+class BelongsToMany extends BelongsToManyBase implements Relation
 {
     use Concerns\BelongsOrMorphsToMany;
+    use Concerns\CanBeCounted;
+    use Concerns\CanBeDetachable;
+    use Concerns\CanBeExtended;
+    use Concerns\CanBePushed;
```

Comment:
Thoughts on the `CanBeX` vs `Xable` naming scheme?

Reply:
Subjective, but I feel this style reads better. ie. relation type can be counted, can be detached, etc.

Reply:
Do we have other examples in our code (or Laravel's) using the CanBe? Just wondering if it's worth breaking convention.

Reply:
Laravel uses the prefix for its concerns, such as [`CanBeOneOfMany`](https://github.com/laravel/framework/blob/11.x/src/Illuminate/Database/Eloquent/Relations/Concerns/CanBeOneOfMany.php) or [`CanBePrecognitive`](https://github.com/laravel/framework/blob/11.x/src/Illuminate/Http/Concerns/CanBePrecognitive.php).

Reply:
Good enough for me!